runtime/databricks/automl_runtime/forecast/pmdarima/model.py

```diff
@@ -0,0 +1,323 @@
+#
+# Copyright (C) 2021 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import List
+
+import pandas as pd
+import mlflow
+import pmdarima
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+
+
+class ArimaModel(mlflow.pyfunc.PythonModel):
+    """
+    ARIMA mlflow model wrapper for univariate forecasting.
+    """
+
+    def __init__(self, pickled_model: bytes, horizon: int, frequency: str,
+                 start_ds: pd.Timestamp, end_ds: pd.Timestamp, time_col: str) -> None:
+        """
+        Initialize the mlflow Python model wrapper for ARIMA.
+        :param pickled_model: the pickled ARIMA model as a bytes object.
+        :param horizon: int number of periods to forecast forward.
+        :param frequency: the frequency of the time series
+        :param start_ds: the start time of training data
+        :param end_ds: the end time of training data
+        :param time_col: the column name of the time column
+        """
+        self._pickled_model = pickled_model
+        self._horizon = horizon
+        self._frequency = frequency
+        self._start_ds = start_ds
+        self._end_ds = end_ds
+        self._time_col = time_col
+        super().__init__()
+
+    def load_context(self, context: mlflow.pyfunc.model.PythonModelContext) -> None:
+        """
+        Loads artifacts from the specified PythonModelContext that can be used by
+        PythonModel.predict when evaluating inputs. When loading an MLflow model with
+        load_pyfunc, this method is called as soon as the PythonModel is
+        constructed.
+        :param context: A PythonModelContext instance containing artifacts that the model
+                        can use to perform inference.
+        """
+        from pmdarima.arima import ARIMA  # noqa: F401
+        return
+
+    def model(self) -> pmdarima.arima.ARIMA:
+        """
+        Deserialize the ARIMA model by pickle.
+        :return: ARIMA model
+        """
+        import pickle
```

Comment:
Could we load the module at the beginning of the file?

Reply:
I was following the style used in the propeht model wrapper ([here](https://github.com/databricks/automl/blob/468206f3161546beeeb5bff1cc1c46d69accc41d/runtime/databricks/automl_runtime/forecast/prophet/model.py#L108)). I probably should have asked earlier whether it's a requirement for the `mlflow.pyfunc.PythonModel`. 
Let me fix both.